### ISSUE TITLE
elasticache module's wait option does nothing

### ISSUE THREAD

**Issue by conlon** [ansible/ansible#23715]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

elasticache module

##### ANSIBLE VERSION

```
ansible 2.3.0.0 (detached HEAD d56ba0945d) last updated 2017/04/14 10:38:39 (GMT -500)
  config file = /Users/mconlon/git/ansible/ansible.cfg
  configured module search path = [u'custom_modules']
  python version = 2.7.12 (default, Oct 11 2016, 05:20:59) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)]
```

##### CONFIGURATION

ansible.cfg:
```
[defaults]
roles_path = roles
lookup_plugins = lookup_plugins
filter_plugins = filter_plugins
hash_behaviour = merge
library = custom_modules
forks = 50

[ssh_connection]
scp_if_ssh = True
pipelining = True
# an attempt to fix ssh unreachable errors when machines are actually reachable
retries = 5
```

##### OS / ENVIRONMENT

running Ansible from OSX Sierra 10.12.3, managing elasticache

##### SUMMARY

using `wait: no` or `wait: false` still results in waiting for the cache cluster result before moving forward with the task.

##### STEPS TO REPRODUCE

Create an elasticsearch cluster with the `wait` option set to no/false and another task following it.

In my code, I'm trying to create multiple clusters, but I end up having to wait for each one to complete before starting the next one:

```yaml
  - name: create elasticache clusters
      elasticache:
        region: "{{ region }}"
        name: "{{ short_env }}-{{ item.name }}"
        engine: "{{ item.engine }}"
        cache_engine_version: "{{ item.cache_engine_version }}"
        cache_parameter_group: "{{ item.cache_parameter_group | default(omit) }}"
        cache_subnet_group: "{{ env }}"
        security_group_ids: "{{ groups.security_groups | map(attribute='group_id') | first }}"

        num_nodes: "{{ item.num_nodes | default(1) }}"
        node_type: "{{ item.node_type }}"
        # don't destory and recreate the cluster to apply these settings
        hard_modify: false
        state: present
        wait: no
      with_items:
        - "{{ vpc.memcached }}"
        - "{{ vpc.redis }}"
```

##### EXPECTED RESULTS

The cluster should start creating and ansible should move on to the next task.

##### ACTUAL RESULTS

The task hangs until a response is received before moving on to the next task

**Comment by ansibot** [comment]
cc @jsdalton @alachaum
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by s-hertel** [comment]
I was unable to recreate this on devel. Your reproducer runs for me and completes before the clusters in the AWS console display the status 'available' (it took minutes more for them to complete). Can you please run your playbook again with extra verbosity (using the `-vvvvvv` flag at the end of the command) and paste it?

**Comment by s-hertel** [comment]
needs_info

**Comment by conlon** [comment]
my bad, this was total user error...I added this option in the wrong file so of course I didn't see any change in behavior 😜 thanks for looking into it and sorry for wasting your time! 😨
